muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/coursepicker/CoursePickerRESTService.java

```diff
@@ -452,7 +453,42 @@ public Response getWorkspace(@PathParam("ID") Long workspaceEntityId) {
       }
     }
     
-    return Response.ok(createRestModel(workspaceEntity, workspace.getName(), workspace.getNameExtension(), workspace.getDescription(), educationTypeName, canSignup, isCourseMember)).build();
+    return Response.ok(createRestModel(workspaceEntity, workspace.getName(), workspace.getNameExtension(), educationTypeName)).build();
+  }
+  
+  @GET
+  @Path("/workspaces/{ID}/signupDetails")
+  @RESTPermit (handling = Handling.INLINE)
+  public Response getSignupDetails(@PathParam("ID") Long workspaceEntityId) {
+    
+    WorkspaceEntity workspaceEntity = workspaceController.findWorkspaceEntityById(workspaceEntityId);
+    if (workspaceEntity == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+    
+    Workspace workspace = null;
+    
+    schoolDataBridgeSessionController.startSystemSession();
+    try {
+      workspace = workspaceController.findWorkspace(workspaceEntity);
+    } finally {
+      schoolDataBridgeSessionController.endSystemSession();
+    }
+
+    if (workspace == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+    
+    String description = workspace.getDescription();
```

Comment:
Workspace description is now fetched from Pyramus but it's faster to retrieve it from Elastic instead. Roughly something like this (warning: coded blind)

```
String description = null;
Iterator<SearchProvider> searchProviderIterator = searchProviders.iterator();
if (searchProviderIterator.hasNext()) {
  SearchProvider searchProvider = searchProviderIterator.next();
  SearchResult searchResult = searchProvider.findWorkspace(workspaceEntity.schoolDataIdentifier());
  if (searchResult.getTotalHitCount() > 0) {
    List<Map<String, Object>> results = searchResult.getResults();
    Map<String, Object> match = results.get(0);
    description = (String) match.get("description");
  }
}